geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandTest.java

```diff
@@ -144,4 +149,110 @@ private void addJvmOptionsForOutOfMemoryErrors(final List<String> commandLine) {
       commandLine.add("-XXexitOnOutOfMemory");
     }
   }
+
+  @Test
+  public void testCreateStartLocatorCommandLine() throws Exception {
+    LocatorLauncher locatorLauncher = new LocatorLauncher.Builder().setMemberName("defaultLocator")
+        .setCommand(LocatorLauncher.Command.START).build();
+
+    String[] commandLineElements = locatorCommands.createStartLocatorCommandLine(locatorLauncher,
+        null, null, new Properties(), null, false, null, null, null);
+
+    Set<String> expectedCommandLineElements = new HashSet<>();
+    expectedCommandLineElements.add(StartMemberUtils.getJavaPath());
+    expectedCommandLineElements.add("-server");
+    expectedCommandLineElements.add("-classpath");
+    expectedCommandLineElements.add(StartMemberUtils.getGemFireJarPath().concat(File.pathSeparator)
+        .concat(StartMemberUtils.CORE_DEPENDENCIES_JAR_PATHNAME));
+    expectedCommandLineElements.add("-Dgemfire.launcher.registerSignalHandlers=true");
+    expectedCommandLineElements.add("-Djava.awt.headless=true");
+    expectedCommandLineElements.add("-Dsun.rmi.dgc.server.gcInterval=9223372036854775806");
+    expectedCommandLineElements.add("org.apache.geode.distributed.LocatorLauncher");
+    expectedCommandLineElements.add("start");
+    expectedCommandLineElements.add("defaultLocator");
+    expectedCommandLineElements.add("--port=10334");
+
+    assertNotNull(commandLineElements);
+    assertTrue(commandLineElements.length > 0);
+    assertEquals(commandLineElements.length, expectedCommandLineElements.size());
+
+    for (String commandLineElement : commandLineElements) {
+      expectedCommandLineElements.remove(commandLineElement);
+    }
+
+    assertTrue(String.format("Expected ([]); but was (%1$s)", expectedCommandLineElements),
+        expectedCommandLineElements.isEmpty());
+  }
+
+  @Test
+  public void testCreateStartLocatorCommandLineWithAllOptions() throws Exception {
```

Comment:
My comments  that follow on StartServerCommandTest apply to this test class as well.

Reply:
@pdxrunner: same as below, the `expectedCommandLineElements` variable should contain the full set of expected elements:

```
    Set<String> expectedCommandLineElements = new HashSet<>();
    expectedCommandLineElements.add(StartMemberUtils.getJavaPath());
    expectedCommandLineElements.add("-server");
    expectedCommandLineElements.add("-classpath");
    expectedCommandLineElements
        .add(StartMemberUtils.getGemFireJarPath().concat(File.pathSeparator).concat(customClasspath)
            .concat(File.pathSeparator).concat(StartMemberUtils.CORE_DEPENDENCIES_JAR_PATHNAME));
    expectedCommandLineElements
        .add("-DgemfirePropertyFile=".concat(gemfirePropertiesFile.getAbsolutePath()));
    expectedCommandLineElements.add(
        "-DgemfireSecurityPropertyFile=".concat(gemfireSecurityPropertiesFile.getAbsolutePath()));
    expectedCommandLineElements.add("-Dgemfire.statistic-sample-rate=1500");
    expectedCommandLineElements.add("-Dgemfire.disable-auto-reconnect=true");
    expectedCommandLineElements.addAll(Arrays.asList(jvmArguments));
    expectedCommandLineElements.add("-Xms".concat(heapSize));
    expectedCommandLineElements.add("-Xmx".concat(heapSize));
    expectedCommandLineElements.add("-XX:+UseConcMarkSweepGC");
    expectedCommandLineElements.add("-XX:CMSInitiatingOccupancyFraction="
        .concat(String.valueOf(StartMemberUtils.CMS_INITIAL_OCCUPANCY_FRACTION)));
    expectedCommandLineElements.add("-Dgemfire.launcher.registerSignalHandlers=true");
    expectedCommandLineElements.add("-Djava.awt.headless=true");
    expectedCommandLineElements.add("-Dsun.rmi.dgc.server.gcInterval=9223372036854775806");
    expectedCommandLineElements.add("-Dgemfire.OSProcess.DISABLE_REDIRECTION_CONFIGURATION=true");
    expectedCommandLineElements.add("org.apache.geode.distributed.LocatorLauncher");
    expectedCommandLineElements.add("start");
    expectedCommandLineElements.add("customLocator");
    expectedCommandLineElements.add("--debug");
    expectedCommandLineElements.add("--force");
    expectedCommandLineElements.add("--hostname-for-clients=localhost");
    expectedCommandLineElements.add("--port=10101");
    expectedCommandLineElements.add("--redirect-output");
```